src/RandMsg.chpl

```diff
@@ -438,6 +438,100 @@ module RandMsg
         }
     }
 
+    inline proc gammaGenerator(kArg: real, ref rs) throws {
+        // TODO fill this in i guess
+        if kArg == 1.0 {
+            // Okay for now im just gonna pick one but we should use method here somehow
+            // maybe not bc zig requires two generators, so inverse is simpler
+            return standardExponentialInvCDF(1, rs)[0];
+        }
+        else if kArg == 0.0 {
+            return 0.0;
+        }
+        else if kArg < 1.0 {  // TODO ask the chpl team about the linter being weird
+            var count = 0;
+            while count < 10000000 {
+                var U = rs.next(0, 1);
+                var V = standardExponentialInvCDF(1, rs)[0];
+                if U <= (1.0 - kArg) {
+                    //can these be more effiection 
+                    var X = U ** (1.0 / kArg);
+                    if X <= V {
+                        writeln("XXX   count is ", count);
+                        return X;
+                    }
+                }
+                else {
+                    var Y = -log((1.0 - U) / kArg);
+                    //can these be more effiection 
+                    var X = (1.0 - kArg + kArg * Y) ** (1.0 / kArg);
+                    if X <= (V + Y) {
+                        writeln("XXX   count is ", count);
+                        return X;
+                    }
+                }
+                count+= 1;
+            }
+            writeln("\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
+            writeln("OMG FIRST LOOP IS BROKE!!");
+
+            return -1.0;  // we failed 10000 times in a row which should practically never happen
+
+        } 
+        else {
+            var b = kArg - 1/3;
```

Comment:
1/3 needs to be 1/3.0 for correct floating point division.  Same for any other computations in this function.  This is
![Screenshot from 2024-12-19 16-25-04](https://github.com/user-attachments/assets/e00486ad-a97a-4f73-9965-0ae84cc50545)
 causing the distribution to be slightly off.